3p/nameframe.max.html

```diff
@@ -3,26 +3,49 @@
 <head>
   <meta charset="utf-8">
   <script>
-    window.onload = function() {
+    var name_data;
+    var writeInBody = false;
+    var instantLoad = false;
+
+    var errorHandler = function(err) {
+      window.parent.postMessage({type: 'no-content', error: err.message}, '*');
+      window.name = '';
+    }
+
+    var render = function() {
       try {
-        var name_data = JSON.parse(window.name);
-        if (!name_data || !name_data['creative']) {
-          throw new Error('Missing or incorrectly formatted JSON ' +
-            'input to nameframe');
-        }
         document.open();
         document.write(name_data['creative']);
         document.close();
         delete name_data['creative'];
         window.name = JSON.stringify(name_data);
-      } catch (err) {
-        window.parent.postMessage(
-          {type: 'no-content', error: err.message}, '*');
-        window.name = '';
+      } catch (e) {
+        errorHandler(e);
       }
     };
+
+    try {
+      name_data = JSON.parse(window.name);
+      writeInBody = name_data['writeInBody'];
```

Comment:
I think this will be more readable if you do:
name_data = JSON.parse(window.name);
if (!name_data || !name_data['creative']) throw new Error('Missing...');
if (writeInBody) {
  ....

Reply:
Done.